src/ff/mod.rs

```diff
@@ -4,6 +4,19 @@
 
 mod field;
 mod prime_field;
+use std::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign};
 
 pub use field::{BinaryField, Field, Int};
 pub use prime_field::{Fp2, Fp31, Fp32BitPrime};
+
+pub trait ArithmeticOps:
```

Comment:
just fyi, I think you can implement `ArithmeticOps` for any `T` (not just field) that implements `Add`, `Sub`, etc by doing something along these lines

```rust
   impl <T> ArithmeticOps for T where T: Add<Output = Self>, ...
```

in this case you don't need to implement it for each field implementation, it will  be derived automatically for it